shopsafe-backend/src/main/java/com/google/sps/data/StoreDatastoreHandler.java

```diff
@@ -0,0 +1,126 @@
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory.Builder;
+
+/*
+ * Class that modifies Store entries in datastore.
+ */
+public class StoreDatastoreHandler {
+    
+    private String id;
+    private Key key;
+    private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    public StoreDatastoreHandler(String id) {
+        this.id = id;
+        this.key = new Builder("Store", id).getKey();
+    }
+
+    /*
+     * Wrapper that handles cases where store is in datastore or not
+     */
+    public void placeStore(Map<String, String[]> ratingsMap) {
+        try {
+            //Found store in datastore
+            Entity existingStoreEntity = datastore.get(key);
+            this.updateStore(existingStoreEntity, ratingsMap); 
+        } catch (Exception EntityNotFoundException) {
+            //Store not in datastore, so create entry
+            this.newStore(ratingsMap);
+        }
+    }
+
+    /*
+     * Creates Rating Entity, does not put in datastore
+     */
+    private Entity createRatingsEntity(Map<String, String[]> ratingsMap) {
+        Entity ratingEntity = new Entity("Rating");
+        ratingsMap.forEach((String ratingField, String[] ratingValue) -> {
+            
+            /* http request getParamMap method formats in array instead of
+             * single value.
+             */
+            ratingEntity.setProperty(ratingField, Double.parseDouble(ratingValue[0]));
+        });
+        return ratingEntity;
+    }
+
+    /*
+     * Places new store along with rating
+     */
+    private void newStore(Map<String, String[]> ratingsMap) {
+        
+        Entity storeEntity = new Entity("Store", id);
+
+        Entity ratingEntity = createRatingsEntity(ratingsMap);
+
+        String ratingEntityIdName = ratingEntity.getKey().getName();
+        ArrayList<String> storeRatingIds = new ArrayList();
+        storeRatingIds.add(ratingEntityIdName);
+        storeEntity.setProperty("ratings", storeRatingIds);
+        
+        datastore.put(ratingEntity);
+        datastore.put(storeEntity);
+    }
+
+    /*
+     * Updates existing store with new rating
+     */
+    private void updateStore(Entity storeEntity, Map<String, String[]> ratingsMap) {
+        Entity ratingEntity = createRatingsEntity(ratingsMap);
+        @SuppressWarnings("unchecked")
+            ArrayList<String> previousRatingIds = (ArrayList<String>) storeEntity
+                .getProperty("ratings");
+            String ratingEntityIdName = ratingEntity.getKey().getName();
+            previousRatingIds.add(ratingEntityIdName);
+            storeEntity.setProperty("ratings", previousRatingIds);
+
+            datastore.put(ratingEntity);
+            datastore.put(storeEntity);
+    }
```

Comment:
newStore and updateStore need more comments. It took me a bit to understand the process by which you are storing the maps.